### PR TITLE
iOS support for Mobile Banking/SCB

### PR DESCRIPTION
**Pull request by ignacionr**
adds UI support for Mobile Banking payments through deep-linking

### CODE REVIEW

File: OmiseSDK/PaymentChooserViewController.swift [hunk 1]
```diff
@@ -70,6 +74,8 @@ extension PaymentChooserOption {
             return [.truemoney]
         case .installmentFirstChoice, .installmentKBank, .installmentKTC, .installmentBBL, .installmentBAY, .installmentSCB:
             return [.installment]
+        case .mobileBankingSCB:
```
Comment (igroomgrim) [inline OmiseSDK/PaymentChooserViewController.swift:77]: Do you need to add `.mobileBankingSCB` case into `func staticIndexPath(forValue value: PaymentChooserOption) -> IndexPath` too.

**Comment by igroomgrim** [comment]
Could you write some test cases in the `ModelTestCase` and `PaymentInformationTestCase` too. 🏁

File: ExampleApp/Shared/PaymentSettingTableViewController.swift [hunk 2]
```diff
@@ -95,6 +95,7 @@ class PaymentSettingTableViewController: UITableViewController {
     @IBOutlet var installmentKTCPaymentCell: UITableViewCell!
     @IBOutlet var installmentKBankPaymentCell: UITableViewCell!
     @IBOutlet var installmentSCBPaymentCell: UITableViewCell!
+    @IBOutlet var mobileBankingSCBPaymentCell: UITableView!
```
Comment (tobias90) [inline ExampleApp/Shared/PaymentSettingTableViewController.swift:98]: Should be `UITableViewCell` and not `UITableView`.

File: OmiseSDK/PaymentChooserViewController.swift (hunk 1 above)
Comment (ignacionr) [reply OmiseSDK/PaymentChooserViewController.swift:77]: Thank you. This is done.

File: ExampleApp/Shared/PaymentSettingTableViewController.swift (hunk 2 above)
Comment (ignacionr) [reply ExampleApp/Shared/PaymentSettingTableViewController.swift]: Updated, thank you!

File: OmiseSDK/Capability.swift [hunk 3]
```diff
@@ -61,6 +61,7 @@ extension Capability {
             case card(Set<CardBrand>)
             case installment(PaymentInformation.Installment.Brand, availableNumberOfTerms: IndexSet)
             case internetBanking(PaymentInformation.InternetBanking)
+            case mobileBanking
```
Comment (tobias90) [inline OmiseSDK/Capability.swift:64]: I assume there are more `mobileBanking` payment methods coming later? So we should probably create a new `PaymentInformation` object and the `case` should look like this:
```
case mobileBanking(PaymentInformation.MobileBanking)
```

**Comment by aquartier** [comment]
Perhaps, I did something wrong but I got this runtime error on my machine. not sure does all code changes on this PR already have successful build. @ignacionr 

```
2020-10-30 14:38:23.081435+0700 ExampleApp[16130:411186] [] nw_protocol_get_quic_image_block_invoke dlopen libquic failed
2020-10-30 14:38:23.353566+0700 ExampleApp[16130:411181] [SDK] Request failed Authentication failure
2020-10-30 14:39:29.404095+0700 ExampleApp[16130:411015] [UI] Payment Chooser: Showing options - Credit Card, InternetBanking, MobileBanking, Alipay, Tesco Lotus, Installment, PromptPay, TrueMoney, CitiPoints
2020-10-30 14:39:29.404403+0700 ExampleApp[16130:411015] [UI] Payment Chooser: Showing options - Credit Card, InternetBanking, MobileBanking, Alipay, Tesco Lotus, Installment, PromptPay, TrueMoney, CitiPoints
2020-10-30 14:39:29.405402+0700 ExampleApp[16130:411015] [UI] Payment Chooser: Showing options - Credit Card, InternetBanking, MobileBanking, Alipay, Tesco Lotus, Installment, PromptPay, TrueMoney, CitiPoints
2020-10-30 14:39:29.430346+0700 ExampleApp[16130:411015] [UI] Payment Chooser: Showing options - Credit Card, InternetBanking, MobileBanking, Alipay, Tesco Lotus, Installment, PromptPay, TrueMoney, CitiPoints
2020-10-30 14:39:29.505995+0700 ExampleApp[16130:411015] *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** __boundsFail: index 12 beyond bounds [0 .. 11]'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff2043a126 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff20177f78 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff204b6cdb _CFThrowFormattedException + 194
	3   CoreFoundation                      0x00007fff204bed58 __boundsFail + 70
	4   CoreFoundation                      0x00007fff2045b52e -[__NSArrayI objectAtIndex:] + 38
	5   UIKitCore                           0x00007fff248b3065 -[UITableViewDataSource tableView:cellForRowAtIndexPath:] + 150
	6   UIKitCore                           0x00007fff23f2be70 -[UITableViewController tableView:cellForRowAtIndexPath:] + 58
	7   OmiseSDK                            0x0000000105666d50 $s8OmiseSDK34AdaptableStaticTableViewControllerC05tableF0_12cellForRowAtSo07UITableF4CellCSo0mF0C_10Foundation9IndexPathVtF + 480
	8   OmiseSDK                            0x0000000105666e65 $s8OmiseSDK34AdaptableStaticTableViewControllerC05tableF0_12cellForRowAtSo07UITableF4CellCSo0mF0C_10Foundation9IndexPathVtFTo + 165
	9   OmiseSDK                            0x0000000105611f5e $s8OmiseSDK28PaymentChooserViewControllerC05tableE0_12cellForRowAtSo07UITableE4CellCSo0lE0C_10Foundation9IndexPathVtF + 334
	10  OmiseSDK                            0x0000000105612265 $s8OmiseSDK28PaymentChooserViewControllerC05tableE0_12cellForRowAtSo07UITableE4CellCSo0lE0C_10Foundation9IndexPathVtFTo + 165
	11  UIKitCore                           0x00007fff248a3545 -[UITableView _createPreparedCellForGlobalRow:withIndexPath:willDisplay:] + 865
	12  UIKitCore                           0x00007fff2486cae3 -[UITableView _updateVisibleCellsNow:] + 2942
	13  UIKitCore                           0x00007fff2488c90e -[UITableView layoutSubviews] + 237
	14  UIKitCore                           0x00007fff24bcd686 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 2874
	15  QuartzCore                          0x00007fff27a5ae67 -[CALayer layoutSublayers] + 258
	16  QuartzCore                          0x00007fff27a61319 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 575
	17  UIKitCore                           0x00007fff24bb8ca1 -[UIView(Hierarchy) layoutBelowIfNeeded] + 573
	18  UIKitCore                           0x00007fff23e2dce5 -[_UISheetPresentationController _sheetLayoutInfoLayout:] + 48
	19  UIKitCore                           0x00007fff23e2ade3 -[_UISheetLayoutInfo _layout] + 360
	20  UIKitCore                           0x00007fff23e2fb6f __54-[_UISheetPresentationController transitionWillBegin:]_block_invoke_2 + 52
	21  UIKitCore                           0x00007fff24bc0131 +[UIView(Animation) performWithoutAnimation:] + 84
	22  UIKitCore                           0x00007fff23e2faac __54-[_UISheetPresentationController transitionWillBegin:]_block_invoke.362 + 120
	23  UIKitCore                           0x00007fff23f6b936 -[_UIViewControllerTransitionCoordinator _applyBlocks:releaseBlocks:] + 294
	24  UIKitCore                           0x00007fff23f67df1 -[_UIViewControllerTransitionContext __runAlongsideAnimations] + 263
	25  UIKitCore                           0x00007fff24bbff5a __63+[UIView(Animation) _setAlongsideAnimations:toRunByEndOfBlock:]_block_invoke + 16
	26  UIKitCore                           0x00007fff24b91790 -[UIViewAnimationState _runAlongsideAnimations] + 24
	27  UIKitCore                           0x00007fff24b90593 -[UIViewAnimationState pop] + 37
	28  UIKitCore                           0x00007fff24b8d9cf +[UIViewAnimationState popAnimationState] + 62
	29  UIKitCore                           0x00007fff24bc19d5 +[UIView _setupAnimationWithDuration:delay:view:options:factory:animations:start:animationStateGenerator:completion:] + 696
	30  UIKitCore                           0x00007fff24bc1e35 +[UIView(UIViewAnimationWithBlocks) animateWithDuration:delay:options:animations:completion:] + 87
	31  UIKitCore                           0x00007fff24b891ef __50-[UITransitionView _startTransition:withDuration:]_block_invoke.164 + 66
	32  UIKitCore                           0x00007fff24bc2302 +[UIView(UIViewAnimationWithBlocks) conditionallyAnimate:withAnimation:layout:completion:] + 76
	33  UIKitCore                           0x00007fff24b88c62 -[UITransitionView _startTransition:withDuration:] + 840
	34  UIKitCore                           0x00007fff24b886bf -[UITransitionView transition:fromView:toView:removeFromView:] + 2890
	35  UIKitCore                           0x00007fff23f63791 -[UIViewControllerBuiltinTransitionViewAnimator animateTransition:] + 1200
	36  UIKitCore                           0x00007fff23f6c505 ___UIViewControllerTransitioningRunCustomTransition_block_invoke_2 + 59
	37  UIKitCore                           0x00007fff240a7599 +[UIInputResponderController _pinInputViewsForInputResponderController:onBehalfOfResponder:duringBlock:] + 99
	38  UIKitCore                           0x00007fff23f6c494 ___UIViewControllerTransitioningRunCustomTransition_block_invoke.641 + 180
	39  UIKitCore                           0x00007fff24bbfe6b +[UIView(Animation) _setAlongsideAnimations:toRunByEndOfBlock:] + 175
	40  UIKitCore                           0x00007fff23f6c31a _UIViewControllerTransitioningRunCustomTransition + 553
	41  UIKitCore                           0x00007fff23e1b53c __56-[UIPresentationController runTransitionForCurrentState]_block_invoke.466 + 2511
	42  UIKitCore                           0x00007fff24b66632 -[_UIAfterCACommitBlock run] + 54
	43  UIKitCore                           0x00007fff2467c179 _runAfterCACommitDeferredBlocks + 333
	44  UIKitCore                           0x00007fff2466c0ac _cleanUpAfterCAFlushAndRunDeferredBlocks + 221
	45  UIKitCore                           0x00007fff2469da70 _afterCACommitHandler + 85
	46  CoreFoundation                      0x00007fff203a76b3 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	47  CoreFoundation                      0x00007fff203a1f3f __CFRunLoopDoObservers + 547
	48  CoreFoundation                      0x00007fff203a24e2 __CFRunLoopRun + 1113
	49  CoreFoundation                      0x00007fff203a1b9e CFRunLoopRunSpecific + 567
	50  GraphicsServices                    0x00007fff2b793db3 GSEventRunModal + 139
	51  UIKitCore                           0x00007fff2466d40f -[UIApplication _run] + 912
	52  UIKitCore                           0x00007fff24672320 UIApplicationMain + 101
	53  ExampleApp                          0x0000000105303e8b main + 75
	54  libdyld.dylib                       0x00007fff20257409 start + 1
	55  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSRangeException', reason: '*** __boundsFail: index 12 beyond bounds [0 .. 11]'
terminating with uncaught exception of type NSException
CoreSimulator 732.17 - Device: iPhone 11 (95CDE1EA-AF51-4D5E-B419-C3B642641E2E) - Runtime: iOS 14.1 (18A8394) - DeviceType: iPhone 11
```

**Comment by igroomgrim** [comment]
If you don't mind, I'll take this PR on my own. 😄

File: OmiseSDK/Capability.swift (hunk 3 above)
Comment (ignacionr) [reply OmiseSDK/Capability.swift:64]: I noticed this was missing from the PR; will update now.

**Comment by igroomgrim** [comment]
Simulator Screen Shot - iPhone X - 2020-11-04 at 11 05 20

**- Updated to Payment methods creator**

Simulator Screen Shot - iPhone X - 2020-11-04 at 11 05 26

**- Supported SCB-MobileBanking source**

Simulator Screen Shot - iPhone X - 2020-11-04 at 11 05 37

**- Merchant application can create SCB-MobileBanking source from Omise iOS SDK**

**Comment by igroomgrim** [comment]
Done, could you guys please help me to review again 🙇

**Review by tookaa2 — APPROVED** [review]

**Review by aquartier — APPROVED** [review]

File: OmiseSDK/Compatibility/OMSConstants.h [hunk 4]
```diff
@@ -17,6 +18,7 @@ extern OMSSourceTypeValue const _Nonnull OMSSourceTypeValueInstallmentKTC;
 extern OMSSourceTypeValue const _Nonnull OMSSourceTypeValueInstallmentKBank;
 extern OMSSourceTypeValue const _Nonnull OMSSourceTypeValueInstallmentSCB;
 extern OMSSourceTypeValue const _Nonnull OMSSourceTypeValueEContext;
+extern OMSSourceTypeValue const _Nonnull OMSSourceTypeValueMobileBankingSCB;
```
Comment (nuxzero) [inline OmiseSDK/Compatibility/OMSConstants.h:21]: I think this is duplicated declaration from above.

Comment (igroomgrim) [reply OmiseSDK/Compatibility/OMSConstants.h]: Removed 🙇

**Review by nuxzero — APPROVED** [review]

**Review by igroomgrim — APPROVED** [review]

**Review by tobias90 — APPROVED** [review]

**Pull request merged**